_content/network/account/implement.md

```diff
@@ -0,0 +1,117 @@
+# Implementing you own Account Server for application management
+
+To implement your own account server, you need to implement a couple of things:
+
+1. An access key validation endpoint
+2. Endpoints to generate and validate application access tokens
+
+# 1. Access Key validation
+
+To validate access keys, MQTT needs to know what rights the access key provides.
+Since the access keys are opaque, it consults an endpoint to fetch those rights:
+
+```plaintext
+GET /api/v2/applications/{app_id}/rights
+Authorization: Key {app_access_key}
+```
+
+If the app id or access key is invalid, it should return `401`.
+
+When successful, the enpoint should return:
+
+```plaintext
+200 OK
+Content-Type: application/json
+
+[
+  "messages:up:r",
+  "messages:down:w"
+]
+```
+
+The meaning of rights returned here are:
+
+- `messages:up:r`:  The right to read the uplink messages from devices an application
+- `messages:down:w`: The right to send the downlink messages to devices of an application
+
+The presence of these rights tells the MQTT plugin what type of messages allow
+the client to see or send.
+
+> Important Note: due to the limited right management of most MQTT implementations, make
```

Comment:
Yes, but the mosquito authentication plugin does not allow us to distinguish between the two.
That's what I'm alluding to.